### PR TITLE
Handmade French translation

### PR DESCRIPTION
**Pull request by christophelachance**
I made a new translation for this module. Every string was manually translated by a native French speaker. Fixes a couple of syntax errors and provides French versions for previously untranslated strings.

### CODE REVIEW

File: language/fr.json [hunk 1]
```diff
@@ -102,66 +102,67 @@
...
         {
-          "label": "Activer \"Solution\""
+          "label": "Activer le bouton \"Solution\""
         }
       ]
     },
     {
-      "label": "Options et textes",
+      "label": "Interface utilisateur",
+      "importance": "low",
```
Comment (otacke) [inline language/fr.json:165]: This field is not required for a translation file, please cmp. https://github.com/otacke/h5p-crossword/blob/master/language/.en.json

File: language/fr.json [hunk 2]
```diff
@@ -172,35 +173,39 @@
...
-          "label": "Texte du bouton \"Vérifier\"",
+          "label": "Étiquette pour le bouton \"Vérifier\"",
           "default": "Vérifier"
         },
         {
-          "label": "Text for \"Submit\" button",
-          "default": "Submit"
+          "label": "Étiquette pour le bouton \"Envoyer\"",
+          "default": "Envoyer"
+
```
Comment (otacke) [inline language/fr.json:182]: Please don't add blank lines.

File: language/fr.json [hunk 3]
```diff
@@ -172,35 +173,39 @@
...
-          "default": "Voir la solution"
+          "label": "Étiquette pour le bouton \"Afficher la solution\"",
+          "default": "Afficher la solution"
         },
         {
-          "label": "Impossible de créer les mots croisés",
-          "default": "Impossible de générer les mots croisés avec les mots donnés. Veuillez réessayer avec moins de mots ou des mots qui ont plus de caractères en commun. Si vous utilisez des mots fixes, assurez-vous que leur position n’empêche pas les autres mots d’être placés."
+          "label": "Impossible de générer le mot-croisé",
+          "default": "Impossible de générer le mot-croisé avec les mots saisis. Veuillez réessayer avec moins de mots ou utiliser des mots qui ont plus de caractères en commun. Si vous utilisez des mots à position fixe, veuillez vous assurer que leur position n'empêche pas les autres mots d'être placés sur la grille."
+
```
Comment (otacke) [inline language/fr.json:195]: Please don't add blank lines.

File: language/fr.json [hunk 4]
```diff
@@ -172,35 +173,39 @@
...
+        
         },
         {
           "label": "Mots problématiques",
-          "description": "Liste les mots qui ont causé des problèmes lors de la génération des mots croisés. @words est un espace réservé qui sera remplacé par les mots problématiques réels.",
+          "description": "Liste des mots qui ont causé des problèmes lors de la génération du mot-croisé. L'identifiant @words sera remplacé par la liste de mots problématiques.",
           "default": "Mot(s) problématique(s): @words"
         },
         {
+
```
Comment (otacke) [inline language/fr.json:208]: Please don't add blank lines.

File: language/fr.json (hunk 3 above)
Comment (otacke) [inline language/fr.json:195]: Please don't add blank lines.

File: language/fr.json [hunk 5]
```diff
@@ -172,35 +173,39 @@
...
+          "label": "Impossible de générer le mot-croisé",
+          "default": "Impossible de générer le mot-croisé avec les mots saisis. Veuillez réessayer avec moins de mots ou utiliser des mots qui ont plus de caractères en commun. Si vous utilisez des mots à position fixe, veuillez vous assurer que leur position n'empêche pas les autres mots d'être placés sur la grille."
+        
         },
         {
-          "label": "Impossible de créer des mots croisés (trop peu de mots)",
-          "default": "Impossible de générer les mots croisés. Il faut au moins deux mots."
+          "label": "Impossible de générer le mot-croisé (trop peu de mots)",
+          "default": "Impossible de générer le mot-croisé. Vous devez saisir au moins deux mots."
+
```
Comment (otacke) [inline language/fr.json:200]: Please don't add blank lines.

**Review by otacke — CHANGES_REQUESTED** [review]
Thanks a lot! The current translation has been contributed by the "Translate H5P community" at https://translate-h5p.tk. Not sure whether they've used some automated translation, but usually some native speaker checks there as well.

I have found a couple of small issues with the structure of the file. Would be nice if you corrected those. If you want to contribute translations by compiling the translation file yourself, it's a good idea to run the file through a service such as https://jsonchecker.com/ in order to check whether the syntax is correct (it's here) and to compare the number of lines of the translation file with the number of lines in the template (the .en.json file). This way, many issues can be spotted before creating the pull request.

You can also try out https://translate-h5p.tk/ if you want to. Those issues that I found in the review would not have happened when using the site's service, and the site also offer a couple of extra features like comparing how others have translated particular strings across all content types, being notified when there are new strings to be translated, etc.

Cheers,
Oliver

**Comment by christophelachance** [comment]
Hello, thanks a lot for taking the time to point out issues and provide useful information. I have removed the blank lines and unused attributes, this new version should be properly formattedm

**Review by otacke — APPROVED** [review]
Excellent!

**Pull request merged**